guide/src/module.md

```diff
@@ -48,53 +53,110 @@ To import the module, either:
 
 ## Documentation
 
-The [Rust doc comments](https://doc.rust-lang.org/stable/book/first-edition/comments.html) of the module
+The [Rust doc comments](https://doc.rust-lang.org/stable/book/ch03-04-comments.html) of the module
 initialization function will be applied automatically as the Python docstring of your module.
 
+For example, building off of the above code, this will print `This module is implemented in Rust.`:
+
 ```python
-import rust2py
+import my_extension
 
-print(rust2py.__doc__)
+print(my_extension.__doc__)
 ```
 
-Which means that the above Python code will print `This module is implemented in Rust.`.
+## Organizing your module registration code
+
+As your Python module grows, it can be helpful to split your Rust code into several Rust modules to
+keep your code readable.
+
+You can pass references to the `PyModule` so that each 
+module registers its own FFI code, which can give your project better 
+[encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)). For example:
+
+```rust,ignore
+// src/lib.rs
+use pyo3::prelude::*;
+
+mod dirutil;
+mod osutil;
+
+#[pymodule]
+fn my_extension(py: Python, m: &PyModule) -> PyResult<()> {
+    self::dirutil::register(py, m)?;
+    self::osutil::register(py, m)?;
+    Ok(())
+}
+
+
+// src/dirutil.rs
+use pyo3::prelude::*;
+
+pub(crate) fn register(py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_class::<SomeClass>()?;
+    Ok(())
+}
+
+#[pyclass]
+struct SomeClass {
+    x: usize,
+};
+
+
+// src/osutil.rs
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+
+pub(crate) fn register(py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_function(wrap_pyfunction!(determine_current_os, m)?)?;
+    Ok(())
+}
+
+#[pyfunction]
+fn determine_current_os() -> String {
+    "linux".to_owned()
+}
+```
 
-## Modules as objects
+## Python submodules
 
-In Python, modules are first class objects. This means that you can store them as values or add them to
-dicts or other modules:
+You can create a module hierarchy within a single extension module by using `PyModule.add_submodule()`.
```

Comment:
Should this cross-reference with Rustdoc magic? I'm not sure how to do that.

Reply:
As in, rustdoc has a way to refer to other parts of the code base and then it will hyperlink for you. I'm not sure how to do it, but I know it exists. 

I'm wondering here if you happen to know if it's possible to cross reference from the book to the API docs?

Reply:
You can just hyperlink to the latest versions with e.g.:
- the pyo3.rs documentation: `https://pyo3.rs/main/doc/pyo3/types/struct.PyModule.html#method.add_submodule`
- docs.rs: https://docs.rs/pyo3/latest/pyo3/types/struct.PyModule.html#method.add_submodule
- 
I'm actually not sure which one you should prefer...fwiw pyo3.rs reflects the current main branch, docs.rs the latest release.

Reply:
The `mdbook` setup has a plugin to substitute in the appropriate link, the following should do it:

`{#PYO3_DOCS_URL}}/pyo3/prelude/struct.PyModule.html#method.add_submodule`